packages/gatsby/content/features/constraints.md

```diff
@@ -214,3 +254,11 @@ gen_enforced_dependency(WorkspaceCwd, DependencyIdent, 'workspace:*', Dependency
 ```
 
 We define a `gen_enforced_dependency` rule that requires the dependency range `workspace:*` to be used if the dependency name is also the name of a valid workspace. The final `workspace_has_dependency` check is there to ensure that this rule is only applied on workspace that currently depend on the specified workspace in the first place (if it wasn't there, the rule would instead force all workspaces to depend on one another).
+
+## Troubleshooting
+
+### `an argument is variable when an instantiated argument was expected`
+
+This error usually means that Prolog expected something, and got another. While it's usually that you passed an unbound value (ie, a variable to fill) to what should be a concrete parameter, it can also be the opposite (a concrete parameter given to something that's supposed to be variable).
+
+Also make sure you're using **single quote strings** when specifying literal strings. In Prolog, double quote strings are a special syntax for lists: `"foo"` is an alias for `['f', 'o', 'o']`, not `'foo'`.
```

Comment:
You can omit the single quotes for the atoms in the list, i.e., you can also write this is as `[f,o,o]`.

For transient data, strings are preferable over atoms, so it could well be that strings may be the better choice in fact.

Reply:
Yep, I added explicit quotes to clarify that double-quotes strings are a composite version of single quote atoms.

As for using actual strings, I tried them in the first constraint implementation, but they were far slower and harder to work with - so we went with atoms. Main issue is around awkwardness of string manipulation (concatenating values, etc), but overall it works fine.